test/ProjectTemplates/Microsoft.Extensions.AI.Templates.IntegrationTests/Snapshots/aichatweb.BasicAspire.verified/aichatweb/aichatweb.Web/Services/SemanticSearch.cs

```diff
@@ -5,14 +5,14 @@ namespace aichatweb.Web.Services;
 
 public class SemanticSearch(
     IEmbeddingGenerator<string, Embedding<float>> embeddingGenerator,
-    IVectorStore vectorStore)
+    VectorStore vectorStore)
 {
     public async Task<IReadOnlyList<IngestedChunk>> SearchAsync(string text, string? documentIdFilter, int maxResults)
     {
         var queryEmbedding = await embeddingGenerator.GenerateVectorAsync(text);
         var vectorCollection = vectorStore.GetCollection<string, IngestedChunk>("data-aichatweb.Web-chunks");
 
-        var nearest = vectorCollection.SearchEmbeddingAsync(queryEmbedding, maxResults, new VectorSearchOptions<IngestedChunk>
+        var nearest = vectorCollection.SearchEmbeddingAsync(queryEmbedding, maxResults, new RecordSearchOptions<IngestedChunk>
```

Comment:
This code generates the embedding manually (line 12 just above) and then passes it in - the nicer way to do this is to have the embedding generator registered in DI as well (or configured with the collection/store), and then you can just call `SearchAsync` here, passing it `text` directly and embedding generation happens internally.

We basically shouldn't need to use ShowEmbeddingAsync unless we specifically want to show low-level usage where embedding generation needs to happen manually for some reason.